app/javascript/shared/components/tags.jsx

```diff
@@ -386,6 +323,74 @@ class Tags extends Component {
       component.forceUpdate();
     }, 100);
   };
+
+  insertSpace(value, position) {
+    return `${value.slice(0, position)} ${value.slice(position, value.length)}`;
+  }
+
+  search(query) {
+    if (query === '') {
+      return new Promise(resolve => {
+        setTimeout(() => {
+          this.resetSearchResults();
+          resolve();
+        }, 5);
+      });
+    }
+
+    return this.index
+      .search(query, {
+        hitsPerPage: 10,
+        attributesToHighlight: [],
+        filters: 'supported:true',
+      })
+      .then(content => {
+        if (this.props.listing === true) {
+          const { additionalTags } = this.state;
+          const { category } = this.props;
+          const additionalItems = (additionalTags[category] || []).filter(
+            t => t.indexOf(query) > -1,
+          );
+          const resultsArray = content.hits;
+          additionalItems.forEach(t => {
+            if (resultsArray.indexOf(t) === -1) {
+              resultsArray.push({ name: t });
+            }
+          });
+        }
+        this.setState({
+          searchResults: content.hits.filter(
+            hit =>
+              !this.selected.includes(hit.name) ||
+              this.selected.includes(hit.name),
```

Comment:
this is the actual fix, correct? can you run me through the logic? we're checking if it both selected and not selected?

Reply:
This is the actual fix. Thinking about it now, it probably makes sense to remove the .filter and its respective block all together. 

I'm really trying to wrap my head around a solution to include the word all the way to the end of the word, but also not have it show up again if it's already been selected. This is something I can work on for the future if it's it is desired. 

I am going to commit these changes now.

Reply:
I'm definitely in favor of doing that in a separate PR, this can be fine too for now! maybe if you could a add a line or two of comments explaining what happens it'd be great!

Reply:
Awesome! 

I made the changes and 2 lines of comments. I'll create a new PR for the other logic once I get it up and running properly! 

Thank you!